## Day 35 — Multiple Linear Regression

## Problem Statement

Predict House Price using:

* Area
* Bedrooms
* Age of House

In [59]:
 # Step 1: Create Dataset

import pandas as pd

data = {
    "Area": [500,700,900,1100,1300,1500,1700,1900],
    "Bedrooms": [1,2,2,3,3,4,4,5],
    "Age": [20,18,15,12,10,8,5,2],
    "Price": [20,30,40,50,60,70,80,90]
}

df = pd.DataFrame(data)

In [51]:
# Self Practice

X = df[['Area', 'Bedrooms', 'Age']]
y = df['Price']

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 42)

from sklearn.linear_model import LinearRegression 
model = LinearRegression()

model.fit(X_train, y_train)

pred = model.predict(X_test)
print(f'Prediction : {pred}')

from sklearn.metrics import mean_absolute_error
mae = mean_absolute_error(pred, y_test)
print(f'MAE : {mae}')

new_df = pd.DataFrame({
    "Area" : [500],
    "Bedrooms" : [1],
    "Age" : [20]
})

prediction = model.predict(new_df)
print(f"\n {new_df} \n Prediction : {prediction}")

print('\n')

print(f"Intercept : {model.intercept_}")
print(f"Coefficient : {model.coef_}")

Prediction : [29.99975001 69.99975001]
MAE : 0.0002499875006272134

    Area  Bedrooms  Age
0   500         1   20 
 Prediction : [20.]


Intercept : -4.993500324983799
Coefficient : [ 0.0499975  -0.00024999 -0.00024999]


In [52]:
# Step 2: Features and Target

X = df[['Area', 'Bedrooms', 'Age']]
y = df['Price']

In [53]:
# Step 3: Train-Test Split

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [54]:
 # Step 4: Train Model

from sklearn.linear_model import LinearRegression

model = LinearRegression()

model.fit(X_train, y_train)

LinearRegression()

In [55]:
# Step 5: Prediction

y_pred = model.predict(X_test)

print(y_pred)

[29.99975001 69.99975001]


In [56]:
# Step 6: MAE

from sklearn.metrics import mean_absolute_error

mae = mean_absolute_error(y_test, y_pred)

print("MAE:", mae)

MAE: 0.0002499875006272134


## Step 7: Check Coefficients

In [57]:
# Step 7: Check Coefficients


print("Intercept:", model.intercept_)
print("Coefficients:", model.coef_)

Intercept: -4.993500324983799
Coefficients: [ 0.0499975  -0.00024999 -0.00024999]


In [58]:
# Step 8: Predict New House

new_house = pd.DataFrame({
    'Area': [1400],
    'Bedrooms': [3],
    'Age': [8]
})

print(model.predict(new_house))

[65.00024999]


## Interview Questions
Answer after running:
1. Why is it called Multiple Linear Regression?

Ans : It is called Multiple Linear Regression because it uses more than one input feature (Area, Bedrooms, Age) to predict a continuous target value (Price).

2. What does a coefficient represent?

Ans : A coefficient represents how much the predicted value changes when that feature increases by 1 unit, while keeping other features constant.

Example : Area coefficient = 0.0499975
Means : If Area increases by 1 unit, the house price increases by about 0.05 units.

3. If Age has a negative coefficient, what does it mean?

Ans : A negative coefficient means that as Age increases, the house price tends to decrease.

Example : Older houses may have lower prices than newer houses.


4. Which is better: MAE = 2 or MAE = 10?

MAE = 2 is better because lower MAE means lower prediction error.

6. How many input features are used in this project?

 Ans : 3 input features

1. Area
2. Bedrooms
3. Age

## Quick Interview Question
 
#### Suppose:

* Area coefficient = 0.05
* If Area increases from 1000 → 1100 (increase of 100),

Approximately how much will the predicted house price increase? 🤔



In [60]:
# Change in Price = Coefficient × Change in Feature

# = 0.05 × 100
# = 5
